src/proxy/authrep.rs

```diff
@@ -146,9 +146,12 @@ pub fn build_call(ar: &AuthRep) -> Result<Request, anyhow::Error> {
 
     let service = ar.service();
 
+    if service.token().is_none() {
+        anyhow::bail!("service token unavailable");
+    }
     let service = Service::new(
         service.id(),
-        threescalers::credentials::Credentials::ServiceToken(service.token().into()),
+        threescalers::credentials::Credentials::ServiceToken(service.token().unwrap().into()),
```

Comment:
This is a redundant - use `let token = if let Some(token) = service.token() { token } else { bail!(...); };` avoiding the double check (`is_none()` + `unwrap()`).

Also, minor nit: typo in commit message `s/if absent/is absent/`.

Reply:
Kinda missed those nits :P